### PR TITLE
Avoid folders which are Reparse Points.

### PR DESCRIPTION
**Pull request by acasetta**
Jarvis fails to index my Start Menu with a System.UnauthorizedAccessException: see [this thread on StackOverflow](https://stackoverflow.com/questions/42124568/how-can-i-access-to-start-menu-path).

This fixes the problem on my end (I am Italian) but I was unable to check if something else breaks.
Be kind: this is my first pull request... 😳

### CODE REVIEW

File: src/Spectre.System/IO/Directory.cs [hunk 1]
```diff
@@ -64,8 +64,22 @@ public IEnumerable<IDirectory> GetDirectories(string filter, SearchScope scope)
         public IEnumerable<IFile> GetFiles(string filter, SearchScope scope)
         {
             var option = scope == SearchScope.Current ? SearchOption.TopDirectoryOnly : SearchOption.AllDirectories;
-            return _directory.GetFiles(filter, option)
-                .Select(file => new File(new FilePath(file.FullName)));
+            IEnumerable<IFile> files = Array.Empty<IFile>();
+            files = files.Concat(_directory.GetFiles(filter, SearchOption.TopDirectoryOnly)
```
Comment (patriksvensson) [inline src/Spectre.System/IO/Directory.cs]: We could change this to: 

```
var files = _directory.GetFiles(filter, SearchOption.TopDirectoryOnly)
    .Select(file => new File(new FilePath(file.FullName)))
```

File: src/Spectre.System/IO/Directory.cs [hunk 2]
```diff
@@ -64,8 +64,22 @@ public IEnumerable<IDirectory> GetDirectories(string filter, SearchScope scope)
...
+                .Select(file => new File(new FilePath(file.FullName))));
+
+            if (option == SearchOption.TopDirectoryOnly)
+            {
+                return files;
+            }
+
+            var directories = _directory.GetDirectories().Where(d => d.Attributes.HasFlag(FileAttributes.ReparsePoint) == false)
+                .Select(dir => new Directory(new DirectoryPath(dir.FullName)));
```
Comment (patriksvensson) [inline src/Spectre.System/IO/Directory.cs:76]: Can we add a line break after this line?

File: src/Spectre.System/IO/Directory.cs [hunk 3]
```diff
@@ -64,8 +64,22 @@ public IEnumerable<IDirectory> GetDirectories(string filter, SearchScope scope)
...
+            foreach (var d in directories)
```
Comment (patriksvensson) [inline src/Spectre.System/IO/Directory.cs]: Can we rename `d` to `directory` here?

File: src/Spectre.System/IO/Directory.cs [hunk 4]
```diff
@@ -64,8 +64,22 @@ public IEnumerable<IDirectory> GetDirectories(string filter, SearchScope scope)
...
+            {
+                files = files.Concat(d.GetFiles(filter, scope));
+            }
+            return files;
```
Comment (patriksvensson) [inline src/Spectre.System/IO/Directory.cs]: Can we add a line break here?

**Review by patriksvensson — CHANGES_REQUESTED** [review]
@acasetta Thank you for this PR, very appreciated!

Left some comments for you that addresses some minor things.

Great work! 👍

File: src/Spectre.System/IO/Directory.cs (hunk 1 above)
Comment (acasetta) [reply src/Spectre.System/IO/Directory.cs]: Mostly. I merged your suggestion:
```
IEnumerable<IFile> files = _directory.GetFiles(filter, SearchOption.TopDirectoryOnly)
    .Select(file => new File(new FilePath(file.FullName)));
```

I have to keep `IEnumerable<IFile>` to honor the method return value.

**Review by patriksvensson — APPROVED** [review]

**Comment by patriksvensson** [comment]
@acasetta This PR has been merged into develop.
Thank you for your contribution! I really appreciate it! 👍

**Pull request merged**